From 3d87d93d0e89ec3657fbd2bfa8f2bb07cba2a630 Mon Sep 17 00:00:00 2001
From: Lukas Czerner <lczerner@redhat.com>
Date: Thu, 23 Oct 2014 09:55:13 -0400
Subject: [fs] ext4: clarify ext4_error message in ext4_mb_generate_buddy_error()

Message-id: <1414058125-4183-160-git-send-email-lczerner@redhat.com>
Patchwork-id: 98264
O-Subject: [RHEL 7.1 PATCH v2 159/171] ext4: clarify ext4_error message in ext4_mb_generate_buddy_error()
Bugzilla: 1150139
RH-Acked-by: Eric Sandeen <sandeen@redhat.com>

From: Theodore Ts'o <tytso@mit.edu>

BZ 1150139
https://bugzilla.redhat.com/show_bug.cgi?id=1150139

BREW 8142710
https://brewweb.devel.redhat.com/taskinfo?taskID=8142710

Upstream commit 94d4c066a4ff170a2671b1a9b153febbf36796f6

 We are spending a lot of time explaining to users what this error
 means.  Let's try to improve the message to avoid this problem.

 Signed-off-by: Theodore Ts'o <tytso@mit.edu>
 Cc: stable@vger.kernel.org
 Signed-off-by: Lukas Czerner <lczerner@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/ext4/mballoc.c b/fs/ext4/mballoc.c
index c8238a2..0a70a35 100644
--- a/fs/ext4/mballoc.c
+++ b/fs/ext4/mballoc.c
@@ -751,8 +751,8 @@ void ext4_mb_generate_buddy(struct super_block *sb,
 
  if (free != grp->bb_free) {
   ext4_grp_locked_error(sb, group, 0, 0,
-          "%u clusters in bitmap, %u in gd; "
-          "block bitmap corrupt.",
+          "block bitmap and bg descriptor "
+          "inconsistent: %u vs %u free clusters",
           free, grp->bb_free);
   /*
    * If we intend to continue, we consider group descriptor
-- 
1.7.1